editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/EventsLog.tsx

```diff
@@ -0,0 +1,170 @@
+            <TableCell>
+              <strong>Session ID</strong>
+            </TableCell>
+            <TableCell></TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {submissions.map((submission) => (
+            <CollapsibleRow {...submission} />
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+};
+
+const CollapsibleRow: React.FC<Submission> = (submission) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  return (
+    <React.Fragment key={submission.eventId}>
+      <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
+        <TableCell
+          sx={{
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+          }}
+        >
+          <Avatar
+            sx={{
+              background: "none",
+              marginRight: (theme) => theme.spacing(1),
+            }}
+          >
+            {submission.eventType === "Pay" ? (
+              <Payment
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            ) : (
+              <Send
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            )}
+          </Avatar>
+          {submission.eventType}
+        </TableCell>
+        <TableCell>
+          {format(new Date(submission.createdAt), "dd/MM/yy hh:mm:ss")}
+        </TableCell>
+        <TableCell>{submission.sessionId}</TableCell>
+        <TableCell>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={() => setOpen(!open)}
+          >
+            {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
+        </TableCell>
+      </TableRow>
+      <TableRow sx={{ background: (theme) => theme.palette.background.paper }}>
+        <TableCell sx={{ paddingBottom: 0, paddingTop: 0 }} colSpan={4}>
+          <Collapse in={open} timeout="auto" unmountOnExit>
+            <FormattedResponse {...submission} />
+          </Collapse>
+        </TableCell>
+      </TableRow>
+    </React.Fragment>
+  );
+};
+
+const FormattedResponse: React.FC<Submission> = (submission) => {
+  if (submission.eventType === "Pay") {
+    return (
+      <Box sx={{ margin: 1 }}>
+        <Typography variant="body2">
+          {`GOV.UK Pay reference: ${submission.eventId}`}
+        </Typography>
+      </Box>
+    );
+  } else {
+    return (
+      <Box
+        component="pre"
+        sx={{
+          fontSize: "1em",
+          margin: 1,
+          maxWidth: "contentWrap",
+          overflowWrap: "break-word",
+        }}
+      >
+        {submission.status === "Success"
+          ? JSON.stringify(JSON.parse(submission.response?.data?.body), null, 2)
+          : JSON.stringify(
+              JSON.parse(submission.response?.data?.message),
+              null,
+              2,
+            )}
+      </Box>
```

Comment:
This is doing a basic if/else right now and grouping all variations of "Send", but we can easily customise responses to each unique event type if desired to emphasise third-party IDs/reference numbers etc. 

For now, I'm just pretty-printing the raw response which is what we most commonly use/share with partners when troubleshooting failures.